src/relayserver/RegistrationManager.ts

```diff
@@ -323,8 +325,10 @@ export class RegistrationManager {
       transactions = transactions.concat(txHash)
     }
     const registerMethod = await this.contractInteractor.getRegisterRelayMethod(this.config.baseRelayFee, this.config.pctRelayFee, this.config.url)
+    const gasLimit = await this.transactionManager.attemptEstimateGas('RegisterRelay', registerMethod, this.managerAddress)
```

Comment:
This won't help: you still estimate had after pending operation. You must wait for it, it out a constant value